example_test.go

```diff
@@ -87,3 +88,17 @@ func Example_structTags() {
 	// Output:
 	// db://
 }
+
+func Example_errorHandling() {
+	var c MyConfig
+	err := config.From("i_dont_exist.txt").FromEnv().To(&c)
+
+	if e, ok := err.(*config.Error); ok {
```

Comment:
This is what I meant by getting rid of the type assertion.

The methods in the switch could easily just be methods that accept an `error` just like the `os` pkg.